subprojects/packagefiles/sdl2_image/meson.build

```diff
@@ -2,12 +2,88 @@ project(
   'sdl2-image',
   'c',
   version: '2.6.3',
-  meson_version: '>=0.49.0',
+  meson_version: '>=0.59.0',
 )
 
-feature_args = ['-DLOAD_BMP']
+feature_args = []
+extra_library_args = []
 
-options = {
+if host_machine.system() == 'windows' and get_option('default_library') != 'static'
+    extra_library_args += '-DDLL_EXPORT'
+endif
+
+# Files that are not directly tied to loaders
+sources = files(
+  'IMG.c',
+  'IMG_stb.c',
+  'IMG_WIC.c',
+)
+
+sdl2_dep = dependency('sdl2')
+
+deps = [sdl2_dep]
+
+# Backends provide support for certain formats, allowing not to depend on external library or use potentially better optimized algorithm
+backend_provided_loaders = []
+
+use_stbimage_backend = get_option('use_stbimage').allowed()
```

Comment:
why not just use a bool here?

Reply:
1. IMO having all backends as features is better than having them as mix of booleans and features.
2. `auto` gained different behavior in later commit.